tests/SpecTests/PrimaryStepDownSpecTest.php

```diff
@@ -0,0 +1,269 @@
+
+        // Send a {replSetStepDown: 5, force: true} command to the current primary and verify that the command succeeded
+        $primary = $this->client->getManager()->selectServer(new ReadPreference(ReadPreference::RP_PRIMARY));
+        $primary->executeCommand('admin', new Command(['replSetStepDown' => 5, 'force' => true]));
+
+        // Retrieve the next batch of results from the cursor obtained in the find operation, and verify that this operation succeeded.
+        $events = [];
+        $observer = new CommandObserver();
+        $observer->observe(
+            function () use ($iterator) {
+                $iterator->next();
+            },
+            function ($event) use (&$events) {
+                $events[] = $event;
+            }
+        );
+        $this->assertTrue($iterator->valid());
+        $this->assertCount(1, $events);
+        $this->assertSame('getMore', $events[0]['started']->getCommandName());
+
+        // Verify that no new connections have been created
+        $this->assertSame($totalConnectionsCreated, $this->getTotalConnectionsCreated());
+
+        // Wait to allow primary election to complete and prevent subsequent test failures
+        $this->waitForMasterReelection();
+    }
+
+    /**
+     * @see https://github.com/mongodb/specifications/tree/master/source/connections-survive-step-down/tests#id10
+     */
+    public function testNotMasterKeepsConnectionPool()
+    {
+        $runOn = [(object) ['minServerVersion' => '4.1.11', 'topology' => [self::TOPOLOGY_REPLICASET]]];
+        $this->checkServerRequirements($runOn);
+
+        // Set a fail point
+        $this->configureFailPoint([
+            'configureFailPoint' => 'failCommand',
+            'mode' => ['times' => 1],
+            'data' => [
+                'failCommands' => ['insert'],
+                'errorCode' => 10107,
+            ],
+        ]);
+
+        $totalConnectionsCreated = $this->getTotalConnectionsCreated();
+
+        // Execute an insert into the test collection of a {test: 1} document.
+        try {
+            $this->insertDocuments(1);
+        } catch (BulkWriteException $e) {
+            // Verify that the insert failed with an operation failure with 10107 code.
+            $this->assertSame(10107, $e->getCode());
+        }
+
+        // Execute an insert into the test collection of a {test: 1} document and verify that it succeeds.
+        $result = $this->insertDocuments(1);
+        $this->assertSame(1, $result->getInsertedCount());
+
+        // Verify that the connection pool has not been cleared
+        $this->assertSame($totalConnectionsCreated, $this->getTotalConnectionsCreated());
+    }
+
+    /**
+     * @see https://github.com/mongodb/specifications/tree/master/source/connections-survive-step-down/tests#id11
+     */
+    public function testNotMasterResetConnectionPool()
+    {
+        $runOn = [(object) ['maxServerVersion' => '4.0.999', 'topology' => [self::TOPOLOGY_REPLICASET]]];
```

Comment:
Since the test states "This test requires a replica set with server version 4.0.", should this also have `minServerVersion: 4.0.0` to ensure we only consider 4.0.x servers?

Reply:
It is skipped implicitly in configureFailpoint (see https://github.com/mongodb/mongo-php-library/blob/e42136525c78020edde72c494cf6c91a2759ab33/tests/FunctionalTestCase.php#L144), but I’ll make it explicit 👍